### PR TITLE
Add callbacks in file manager

### PR DESCRIPTION
**Pull request by Ngoguey42**
On top of #1867

### CODE REVIEW

**Comment by Ngoguey42** [comment]
@zshipko If you're working on the dict problems. This PR implements some missing features that will surely help

File: src/irmin-pack/unix/file_manager.ml [hunk 1]
```diff
@@ -29,12 +29,20 @@ module Make
...
+  type suffix_consumer_data = { after_flush : unit -> unit }
+
   type t = {
     dict : Dict.t;
     control : Control.t;
     suffix : Suffix.t;
     index : Index.t;
     use_fsync : bool;
+    mutable dict_consumers : dict_consumer_data list;
+    mutable suffix_consumers : suffix_consumer_data list;
```
Comment (icristescu) [inline src/irmin-pack/unix/file_manager.ml:45]: why are these lists?

Comment (Ngoguey42) [reply src/irmin-pack/unix/file_manager.ml:45]: `suffix_consumers` will be of length 3, for the node/contents/commit stores.

`dict_consumers` will be of length 1, for the dict module. An `option` could be used here, but I picked a list for symmetry with `suffix_consumers`.

**Review by icristescu — APPROVED** [review]
Thanks!

**Pull request merged**